### PR TITLE
Update pipeline structure to fetch 2023 data

### PR DESCRIPTION
**Pull request by ceenell**
Changes made:
-----------
This takes steps to convert the data processing pipeline into `targets`. I'm using `snotelr` to fetch SNOTEL data, and adapting prior scripts into modular functions. This ends at the data processing stage, with a function added to calculate the SWE magnitude variables. Additional changes are needed to calculate timing metrics and incorporate the svg pre-processing into `targets`. 

Testing:
----------------------------
Before making this pull request, I:
- [ ] Cleaned the code the way Vue likes it - run 'npm run lint --fix'        
- [ ] Made sure all tests run
- [ ] Ran WAVE plugin 508 compliance tool

I can confirm this has been checked on:
- [ ] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox
- [ ] Samsung Internet
- [ ] Internet Explorer 11 (not supported, but still needs at least a working user redirect page)

### CODE REVIEW

**Comment by aaarcher-usgs** [comment]
For me to run this, I had to make a *new* RStudio project in the "data_processing_pipeline" folder and had to create the empty "data_processing_pipeline/1_fetch/out" folder. 

I think since this is a pipeline, and there's an existing RStudio `.Rproj` file in the main directory, maybe moving `_targets.R` out of the `data_processing_pipeline` folder would work better to use existing project. 

Additionally, adding an empty `out/` folder is needed.

File: data_processing_pipeline/1_fetch.R [hunk 1]
```diff
@@ -0,0 +1,43 @@
+source('1_fetch/src/fetch_SNOTEL_data.R')
+
+p1_targets_list <- list(
+
+  # date last updated
+  tar_target(
+    today,
+    as.Date('2023-03-29')
```
Comment (aaarcher-usgs) [inline data_processing_pipeline/1_fetch.R:8]: I didn't change this date, just used this March value.

**Review by aaarcher-usgs — APPROVED** [review]
Other than considering where `_targets.R` is relative to the `.Rproj` file, (either changing targets to be out of the sub-directory *or* change the .Rproj to inside it) and adding an empty "out" folder, this ran successfully in ~30 minutes.

**Review by jpadilla-usgs — APPROVED** [review]
Howdy Cee,

I approve this PR after the inline comments are addressed and this comment from Althea is addressed:
> For me to run this, I had to make a new RStudio project in the "data_processing_pipeline" folder and had to create the empty "data_processing_pipeline/1_fetch/out" folder.

We should definitely add that bit of file structure in there.

As to Althea's second comment:
> I think since this is a pipeline, and there's an existing RStudio .Rproj file in the main directory, maybe moving _targets.R out of the data_processing_pipeline folder would work better to use existing project.


I also had to do the same (I added the Rproj file to at the top level and then `cd`'d into the proper `wd`). I always add an Rproj and do think that should be in there. The `README` could use an update for running this thing locally - it could be just a sentence or two under the `Project Setup` header. I recommend this, because running targets in this repo is not immediately obvious (e.g., the process is _not_ "open R.proj and run `targets::tar_make()`")

File: data_processing_pipeline/2_process.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
+source('2_process/src/prep_SNOTEL.R')
+
+p2_targets_list <- list(
+  # parse data
+  tar_target(
+    snotel_data,
+    snotel_files |> purrr::map_dfr(read_csv) |>
+      mutate(year = year(date),
+             WY = dataRetrieval::calcWaterYear(date))
```
[the remark this answers is not in the document]
Comment (ceenell) [reply data_processing_pipeline/2_process.R:9]: `calcWaterYear` is clutch!

**Comment by ceenell** [comment]
Thanks @aaarcher-usgs and @jpadilla-usgs! I have addressed the comments.

**Pull request merged**